test/unit/sync_checker/checks/details_check_test.rb

```diff
@@ -0,0 +1,145 @@
+require 'minitest/autorun'
+require 'mocha/setup'
+require 'active_support'
+require 'active_support/json'
+require 'active_support/core_ext'
+
+require_relative '../../../../lib/sync_checker/checks/details_check'
+
+module SyncChecker::Checks
+  class DetailsCheckTest < Minitest::Test
+    def test_non_200_returns_an_empty_array
+      response = stub(
+        response_code: 404,
+        body: {
+        }.to_json
+      )
+
+      expected = {
+        body: "a bit of html"
+      }
+
+      assert_equal [], DetailsCheck.new(expected).call(response)
```

Comment:
I'm going to merge this so as we can get some feedback by running it. We have a couple of other checks to add before using it to make any migration 'decisions' in any case so I'll add the status check to the list.